### ISSUE TITLE
Legacy Clients

### ISSUE THREAD

**Issue by RaviMatharu** [MicrosoftDocs/OfficeDocs-Exchange#211]
I wanted to check if these rules can be used to block both Outlook 2010 and Outlook 2016  clients when accessing from outside the corporate network?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b9cdf39f-da71-43ab-0ceb-a816bdc19d04
* Version Independent ID: 1fff6b28-f8d8-d9b3-1730-db2cac7835a0
* Content: [Client Access Rules in Exchange Online](https://docs.microsoft.com/en-us/exchange/clients-and-mobile-in-exchange-online/client-access-rules/client-access-rules)
* Content Source: [Exchange/ExchangeOnline/clients-and-mobile-in-exchange-online/client-access-rules/client-access-rules.md](https://github.com/MicrosoftDocs/OfficeDocs-Exchange/blob/public/Exchange/ExchangeOnline/clients-and-mobile-in-exchange-online/client-access-rules/client-access-rules.md)
* Service: **exchange-online**
* Product: **exchange**
* GitHub Login: @chrisda
* Microsoft Alias: **chrisda**

**Comment by chrisda** [comment]
@RaviMatharu, yes you can use client access rules to block external Outlook clients. A couple of points:

- I'm not sure if Outlook 2010 is even a supported client for Exchange Online anymore, but conceptually, any Outlook client that uses Outlook Anywhere (RPC over HTTP) or MAPI over HTTP (MAPI/HTTP) can be blocked.

- The feature itself doesn't seem to care specifically about internal vs. external clients; it simply blocks client connections based on the settings you configure, which could be internal, external, or both. From the topic:

  Connections from your local network aren't automatically allowed to bypass Client Access Rules. Therefore, when you create Client Access Rules that block client connections to Exchange Online, you need to consider how connections from your internal network might be affected. The preferred method to allow internal client connections to bypass Client Access Rules is to create a highest priority rule that allows client connections from your internal network (all or specific IP addresses). That way, the client connections are always allowed, regardless of any other blocking rules that you create in the future.

If this answers your question, please close the issue. Thanks.

**Comment by AndreaBarr** [comment]
@RaviMatharu It looks like Chris was able to answer your questions. I am going to close this issue as resolved. If this doesn't answer your questions please re-open this issue.

**Comment by RaviMatharu** [comment]
So would these rules work?



New-ClientAccessRule -Name AllowAccessInternal -Action AllowAccess -AnyOfAuthenticationTypes ADFSAuthentication,BasicAuthentication,NonBasicAuthentication -AnyOfClientIPAddressesOrRanges <InternalAddresses> -Enabled $true -Priority 10


New-ClientAccessRule -Name BlockAccessExternal -Action AllowAccess -AnyOfAuthenticationTypes ADFSAuthentication,BasicAuthentication,NonBasicAuthentication -ExceptAnyOfClientIPAddressesOrRanges <InternalAddresses> -AnyOfProtocols IMAP4,OutlookAnywhere,POP3 -Enabled $true -Priority 20

**Comment by AndreaBarr** [comment]
@chrisda Would the above rules work for RaviMatharu?

**Comment by chrisda** [comment]
@RaviMatharu: here's how I think it breaks down. I assume the goal is to block POP3, IMAP4, and Outlook Anywhere for external users, but allow for internal users (which assumes the AnyOfClientIPAddressesOrRanges and ExceptAnyOfClientIPAddressesOrRanges parameters define your internal IP addresses.)

The first rule matches for client connections that use the specified authentication types AND whose IP addresses are in the specified range (both).

    - Internal POP3, IMAP4, Outlook anywhere users would skip the first rule, because according to the [table in the topic](https://docs.microsoft.com/Exchange/clients-and-mobile-in-exchange-online/client-access-rules/client-access-rules#authentication-types-and-protocols) those authentication types aren't even used with those protocols. If they're using some other client with one of the specified authentication types, they're allowed and rule processing stops.

    - External POP3, IMAP4, Outlook Anywhere users would skip the first rule based on the authentication method problem AND because their IP addresses aren't in the internal range.

Is the Action in the second rule supposed to be BlockAccess? As it's written, the rule allows access if the client uses the specified authentication type AND the specified protocol (both) UNLESS their IP address is defined in the specified range.

    - Internal POP3, IMAP4, Outlook Anywhere users won't match the second rule because of the authentication method AND the internal IP address range exception. Even if they matched the authentication methods AND protocol, they would be excused by the IP address range exception.

    - External POP3, IMAP4, Outlook Anywhere users won't match the second rule because of the combination of authentication method AND protocol. They match the protocol, but not the authentication method.

Final thoughts:

- Test your rules. See [Use Exchange Online PowerShell to test Client Access Rules](https://docs.microsoft.com/Exchange/clients-and-mobile-in-exchange-online/client-access-rules/procedures-for-client-access-rules#use-exchange-online-powershell-to-test-client-access-rules).

- If it was me, I would de-couple the combination of authentication type and protocol for POP3, IMAP4, and Outlook Anywhere in the rules. Use the [table in the topic](https://docs.microsoft.com/Exchange/clients-and-mobile-in-exchange-online/client-access-rules/client-access-rules#authentication-types-and-protocols) and only combine authentication type and protocol in the same rule where it's actually supported (ActiveSync, EAC, OWA, or PowerShell).

  - Create two separate higher priority Allow rules for your internal IP addresses: one to allow the authentication methods you want, the other to allow POP3, IMAP4, and Outlook Anywhere. In other words, one rule conditions are AnyOfAuthenticationTypes and AnyOfClientIPAddressesOrRange. The second rule conditions are AnyOfProtocols and AnyOfClientIPAddressesOrRange. External users will always skip these allow rules based on their IP addresses.

    - Create two separate lower priority Block rules for external users: one to block the authentication methods you want, the other to block POP3, IMAP4, and Outlook Anywhere. In other words, one rule condition is AnyOfAuthenticationTypes. The second rule condition is AnyOfProtocols. You don't need to mess with IP address exceptions (although you can include them if it makes you feel better), because internal users would have already matched one of the higher priority Allow rules, and won't reach these lower priority Block rules.

    - Alternatively, you could create two separate Block rules: one to block the authentication methods you want, the other to block POP3, IMAP4, and Outlook Anywhere, where both rules exempt internal IP addresses. In other words, one Block rule condition is AnyOfAuthenticationTypes with the ExceptAnyOfClientIPAddressesOrRanges to exempt your internal IP address. The other rule condition is AnyOfProtocols with the ExceptAnyOfClientIPAddressesOrRanges to exempt your internal IP address. Internal users aren't blocked by either rule because of their IP addresses.
